meson.build

```diff
@@ -10,7 +10,14 @@ project('icecream-sundae', 'cpp',
 
 icecc = dependency('icecc')
 glib = dependency('glib-2.0')
-ncurses = dependency('ncursesw')
+
+if (build_machine.system() == 'linux')
+    ncurses = dependency('ncursesw')
+    ncurses_cxxflag = ''
+else
+    ncurses = dependency('ncurses')
+    ncurses_cxxflag = '-DNCURSES_WIDECHAR'
+endif
```

Comment:
Could we do this more dynamically:

{code}
ncurses = dependency('ncursesw', required: false)
ncurses_cxxflags = ''
if not ncurses.found()
    ncurses = dependency('ncurses')
    ncurses_cxxflag = '-DNCURSES_WIDECHAR'
endif

Reply:
I simplified that even further to 

```meson
ncurses = dependency('ncursesw')
ncurses_cxxflag = '-DNCURSES_WIDECHAR'
```

after realizing that the `pkgconfig` provided by `brew` on macOS actually has a configuration for `ncursesw`. Unfortunately the `NCURSES_WIDECHAR` definition still needs to be provided.